homeassistant/components/binary_sensor/ring.py

```diff
@@ -0,0 +1,107 @@
+
+from homeassistant.loader import get_component
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_ENTITY_NAMESPACE, CONF_MONITORED_CONDITIONS,
+    ATTR_ATTRIBUTION)
+
+DEPENDENCIES = ['ring']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Sensor types: Name, category, device_class
+SENSOR_TYPES = {
+    'ding': ['Ding', ['doorbell'], 'occupancy'],
+    'motion': ['Motion', ['doorbell'], 'motion'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE):
+        cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for a Ring device."""
+    ring = get_component('ring')
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        for device in ring.RING.data.doorbells:
+            if 'doorbell' in SENSOR_TYPES[sensor_type][1]:
+                sensors.append(RingBinarySensor(hass,
+                                                device,
+                                                sensor_type))
+    add_devices(sensors, True)
+    return True
+
+
+class RingBinarySensor(BinarySensorDevice):
+    """A binary sensor implementation for Ring device."""
+
+    def __init__(self, hass, data, sensor_type):
+        """Initialize a sensor for Ring device."""
+        super(RingBinarySensor, self).__init__()
+        self._cache = hass.config.path(DEFAULT_CACHEDB)
+        self._sensor_type = sensor_type
+        self._data = data
+        self._name = "{0} {1}".format(self._data.name,
+                                      SENSOR_TYPES.get(self._sensor_type)[0])
+        self._device_class = SENSOR_TYPES.get(self._sensor_type)[2]
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return True if the binary sensor is on."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the class of the binary sensor."""
+        return self._device_class
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {}
+        attrs[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
+
+        attrs['device_id'] = self._data.id
+        attrs['firmware'] = self._data.firmware
+        attrs['timezone'] = self._data.timezone
+
+        if self._data.alert and self._data.alert_expires_at:
+            attrs['expires_at'] = self._data.alert_expires_at
+            attrs['state'] = self._data.alert.get('state')
+
+        return attrs
+
+    def update(self):
+        """Get the latest data and updates the state."""
+        self._data.check_alerts(cache=self._cache)
```

Comment:
This will be used to save the information about a notification received by the doorbell.

I will test it using the shared hass.data to check it if is still required.